web-bundle/src/main/java/com/graphhopper/replica/StableEdgeIdManager.java

```diff
@@ -0,0 +1,40 @@
+package com.graphhopper.replica;
+
+import com.graphhopper.GraphHopper;
+import com.graphhopper.routing.util.AllEdgesIterator;
+import com.graphhopper.routing.util.EncodingManager;
+import com.graphhopper.stableid.StableIdEncodedValues;
+import com.graphhopper.storage.GraphHopperStorage;
+import com.graphhopper.storage.NodeAccess;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class StableEdgeIdManager {
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final GraphHopper graphHopper;
+
+    public StableEdgeIdManager(GraphHopper graphHopper) {
+        this.graphHopper = graphHopper;
+    }
+
+    public void setStableEdgeIds() {
+        GraphHopperStorage graphHopperStorage = graphHopper.getGraphHopperStorage();
+        AllEdgesIterator edgesIterator = graphHopperStorage.getAllEdges();
+        NodeAccess nodes = graphHopperStorage.getNodeAccess();
+        EncodingManager encodingManager = graphHopper.getEncodingManager();
+        StableIdEncodedValues stableIdEncodedValues = StableIdEncodedValues.fromEncodingManager(encodingManager);
+
+        // Set both forward and reverse stable edge IDs for each edge
+        int assignedIdCount = 0;
+        while (edgesIterator.next()) {
+            // Ignore setting stable IDs for transit edges, which have a distance of 0
```

Comment:
When/how does this occur out of curiosity?

Reply:
This just happens to be what GraphHopper does by default with transit edges (assign them a distance of 0). Currently, that's been the best way I've figured out for how to distinguish between transit edges (which we don't want to assign IDs) and non-transit edges (which we do), so it's what I've been using as a check for right now.